libndt.hpp

```diff
@@ -1960,8 +2017,37 @@ bool Client::ndt7_upload() noexcept {
     constexpr auto measurement_interval = 0.25;
     std::chrono::duration<double> interval = now - latest;
     if (interval.count() > measurement_interval) {
+      nlohmann::json measurement;
+      measurement["AppInfo"] = nlohmann::json();
+      measurement["AppInfo"]["ElapsedTime"] = elapsed.count();
+      measurement["AppInfo"]["NumBytes"] = total;
+#ifdef __linux__
+      // Read tcp_info data for the socket and print it as JSON.
+      struct tcp_info tcpinfo{};
+      socklen_t tcpinfolen = sizeof(tcpinfo);
+      if (sys_getsockopt(sock_, IPPROTO_TCP, TCP_INFO, (void *)&tcpinfo,
+                         &tcpinfolen) == 0) {
+        measurement["TCPInfo"] = nlohmann::json();
+#define XX(lower_, upper_) measurement["TCPInfo"][#upper_] = (uint64_t)tcpinfo.lower_;
+        NDT7_ENUM_TCP_INFO
+#undef XX
+      }
+#endif  // __linux__
       on_performance(nettest_flag_upload, 1, static_cast<double>(total),
                      elapsed.count(), max_upload_time);
+      // This could fail if there are non-utf8 characters. Since there are just
+      // ASCII characters, we do not handle exceptions here.
```

Comment:
I'd say "just integers in this structure"

Reply:
That sentence also applies to the keys ("TCPInfo" and "AppInfo") though, which are strings and guaranteed to only use the ASCII set.

Reply:
You're right, I missed those. So, the original sentence is definitely more accurate!